### PR TITLE
wip: forloop skip failures -- very first pass

### PR DESCRIPTION
**Pull request by sqwishy**

### CODE REVIEW

File: backend/src/worker_flow.rs [hunk 1]
```diff
@@ -159,24 +163,28 @@ pub async fn update_flow_status_after_job_completion(
...
-                    )
-                    .fetch_optional(db)
-                    .await?
-                    .flatten();
-                    results.push(result.clone());
-                }
+                use futures::TryStreamExt;
+                let results = sqlx::query_as(
+                    "
+                  SELECT result
```
Comment (sqwishy) [inline backend/src/worker_flow.rs:175]: I forgot to add an ORDER BY clause here.

File: backend/src/worker_flow.rs [hunk 2]
```diff
@@ -159,24 +163,28 @@ pub async fn update_flow_status_after_job_completion(
...
+                    FROM completed_job
+                   WHERE id = ANY($1)
+                     AND workspace_id = $2
+                     AND success = true
```
Comment (rubenfiszel) [inline backend/src/worker_flow.rs:179]: does your autoformatter does this automatically ? this is nice

File: backend/src/worker_flow.rs [hunk 3]
```diff
@@ -91,11 +91,15 @@ pub async fn update_flow_status_after_job_completion(
...
+    let skip_loop_failures = skip_loop_failures(flow, old_status.step, &mut tx)
+        .await?
+        .unwrap_or(false);
+
     let (step_counter, new_status) = match &old_status.modules[old_status.step as usize] {
         module_status @ FlowStatusModule::InProgress {
             iterator: Some(Iterator { index, itered, .. }),
             ..
-        } if (index.to_owned() as usize) < itered.len() - 1 && success => {
+        } if (index.to_owned() as usize) < itered.len() - 1 && (success || skip_loop_failures) => {
```
Comment (rubenfiszel) [inline backend/src/worker_flow.rs:102]: I think you might be missing the case where it's the last iteration and that iteration failed, in which case you want to have a consistent behavior and still consider the module to be a success no matter what, which you need to adjust at line 113

**Comment by rubenfiszel** [comment]
I will take over this PR for integration with the UI. Thanks for your contribution!

File: backend/src/worker_flow.rs (hunk 2 above)
Comment (sqwishy) [reply backend/src/worker_flow.rs:179]: No, I'm afraid I do this manually with my fingers :)

**Comment by rubenfiszel** [comment]
Superseeded by #258

**Pull request closed without merging**